from pyrogram import Client, filters
from pyrogram.errors import UserAlreadyParticipant
import asyncio
from helpers.decorators import authorized_users_only, errors
from callsmusic.callsmusic import client as USER
from config import SUDO_USERS


@Client.on_message(filters.command(["asistan"]) & ~filters.private & ~filters.bot)
@authorized_users_only
@errors
async def addchannel(client, message):
    chid = message.chat.id
    try:
        invitelink = await client.export_chat_invite_link(chid)
    except:
        await message.reply_text(
            "<b>Ã¶nce beni yÃ¶netici yap!</b>",
        )
        return

    try:
        user = await USER.get_me()
    except:
        user.first_name = "helper"

    try:
        await USER.join_chat(invitelink)
        await USER.send_message(message.chat.id, "ğŸ¤–: sesli sohbette mÃ¼zik Ã§almak iÃ§in buraya katÄ±ldÄ±m")
    except UserAlreadyParticipant:
        await message.reply_text(
            "<b>sohbetinizde zaten yardÄ±mcÄ±yÄ±m</b>",
        )
    except Exception as e:
        print(e)
        await message.reply_text(
            f"<b>ğŸ›‘ Flood Wait Error ğŸ›‘ \n User {user.first_name} userbot iÃ§in yoÄŸun katÄ±lÄ±m istekleri nedeniyle grubunuza katÄ±lamadÄ±m! KullanÄ±cÄ±nÄ±n grupta yasaklÄ± olmadÄ±ÄŸÄ±ndan emin olun."
            "\n\nVeya Grubunuza el ile Asisstant ekleyin ve yeniden deneyin</b>",
        )
        return
    await message.reply_text(
        "<b>yardÄ±mcÄ± userbot sohbetinize katÄ±ldÄ±</b>",
    )


@USER.on_message(filters.group & filters.command(["userbot"]))
@authorized_users_only
async def rem(USER, message):
    try:
        await USER.leave_chat(message.chat.id)
    except:
        await message.reply_text(
            f"<b>KullanÄ±cÄ± grubunuzdan ayrÄ±lamadÄ±! Floodwaits olabilir."
            "\n\nYa da beni manuel olarak grubunuza tekmelersiniz.</b>",
        )
        return
    
@Client.on_message(filters.command(["asistanayril"]))
async def bye(client, message):
    if message.from_user.id in SUDO_USERS:
        left=0
        failed=0
        lol = await message.reply("Asistan TÃ¼m sohbetleri bÄ±rakma")
        async for dialog in USER.iter_dialogs():
            try:
                await USER.leave_chat(dialog.chat.id)
                left = left+1
                await lol.edit(f"Asistan ayrÄ±lÄ±yor... Left: {left} chats. Failed: {failed} chats.")
            except:
                failed=failed+1
                await lol.edit(f"Asistan ayrÄ±lÄ±yor... Left: {left} chats. Failed: {failed} chats.")
            await asyncio.sleep(0.7)
        await client.send_message(message.chat.id, f"Left {left} chats. Failed {failed} chats.")

# Idon'tknowwhatisthis
